### DISCUSSION TITLE
Lighthouse deck doesn’t receive correctly after take off

### DISCUSSION THREAD

**Discussion by Shibaandcorgii** [bitcraze/discussions#71] [Q&A]
Hi! We have a problem with stability of crazyflie+lighthouse deck. Before we take off, we have checked lighthouse status and all of them are green and the led on lighthouse deck is green too , then when we take off with cfclient or skybrush live then the led is turn to orange that make our drone unstable and crash on the floor . We wonder what is the cause of this problem?

**Comment by knmcguire** [comment]
Hi!

We would need some more information about your system and firmware: 

- Connect your cfclient to the CFclient, go to 'About' menu item and copy the content of the 'debug' tab here. 
- Moreover, copy the content of the console tab (view->tabs) while connected to the crazyflie as well,

**Comment by Shibaandcorgii** [comment]
This is content on debug
![8645BCA2-7585-4E7F-A705-52C39D8D395A](https://user-images.githubusercontent.com/106365446/171587708-5fc11f50-d3ed-4aea-9c9d-7e340895e0b8.jpeg)

This is content on console tap
![93B01A06-00DC-46DE-BA80-72A594B9A019](https://user-images.githubusercontent.com/106365446/171587727-624c2b79-c7e7-4cbc-881b-1eeae2ba3603.jpeg)

Thanks for your kindness.

**Reply by knmcguire** [reply]
Great ! thanks for this. Now I know that the lighthouse deck has been correctly updated. 
Just some additional questions:


- Could you also share a screenshot of the lighthouse positioning tab too with the LH pos on. Also a screenshot of the 'manage geometery' dialog gives us information about how the basestations are configured. 
- And now how high does the crazyflie fly in your flight area? 
- Would it be possible to [flash the latest release](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/#firmware-upgrade) on the crazyflie too and try that out? I can see that the firmware version you are flying with has been modified a lot.

**Comment by Shibaandcorgii** [comment]
• lighthouse positioning tab
![BDEA91DB-EB2F-492B-909B-6CCE215AC757](https://user-images.githubusercontent.com/106365446/171593921-0435a613-75b2-4a49-bae9-34c68d17fd91.jpeg)

• manage geometry
![F8E263EA-55B4-4206-AFC0-DD6AF458D3FC](https://user-images.githubusercontent.com/106365446/171594528-6c6b23fb-0b1f-48e7-8d20-66f93d105244.jpeg)

• We try take off heigh as a default in cfclient and also skybrush live (~1m) 
Note: we just take off with take off tab in each program

• We flashed latest release firmware again and it’s the same result as video in this link.
https://drive.google.com/file/d/1K3z3eqg6XKqnGseRY4nJ8WsPb777qFVC/view?usp=drivesdk

**Reply by knmcguire** [reply]
Ah great! Thanks for the video and the screenshots. 

- The basestations are not too far apart so that is not it
- There might be reflections in the environment... but then you would see the orange led and green led flickering so I don't think that is it.

**Comment by knmcguire** [comment]
I noticed in the video you send before that you have a propguard. Low voltages and bad batteries could cause the lighthouse deck to not get enough power. 

Have you tried flying without the propguard and a fresh, full battery? It might be that it's too heavy, and that the power to the motors do not leave enough for the lighthouse deck. 

@tobbeanton does this sound familiar?

**Reply by Shibaandcorgii** [reply]
This is lighthouse positioning tab before we take off.
![71E8240E-DF6C-4A53-ABE4-F64E9694E5F0](https://user-images.githubusercontent.com/106365446/171783379-34061f06-4c22-4ffb-95b6-2f55bb900304.jpeg)


We tried take off without propeller guard and the result is that the led isn’t orange when take off but it’s unstable like this video link.
https://drive.google.com/file/d/1P43nwMnw4ADKaYJOQj37BDhzDOwFyYtQ/view?usp=drivesdk

and we also tried to fly with our simple design  drone show.
https://drive.google.com/file/d/1UEBNA0egTWoEJSmOEgP4Q_TQzjxlM-3V/view?usp=drivesdk

**Comment by krichardsson** [comment]
I think you should start by checking two things: the  base station geometry and reflective surfaces in the room
1. If the base station geometry is not correct, the Crazyflie will get different positions from the two base stations which will cause confusion in the estimator. You can verify if this is the problem by first looking at the numbers in your geometry to if they look reasonable, are the base stations at 2.13 and 2.03 meters height? Secondly you can do an experiment and turn off one of the base stations and fly with only one. The positioning will not be as good, but should generally work. With only one base station there is no competing information and if the root cause of the problem is the geometry, it should work better.
2. Reflections are problematic as the Crazyflie can not distinguish the reflection from the direct light from a base station. Check if you have any items in the room that might reflect the light from the base stations, like windows, mirrors, shiny metal surfaces or similar. Try to cover them to see if it makes a difference.

For the geometry you can try to run the more advanced geometry estimation script (see https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/lighthouse/multi_bs_geometry_estimation.py and https://www.bitcraze.io/2022/03/multiple-base-stations-with-lighthouse/) it should provide a better geometry estimation.
